plugins/node/opentelemetry-instrumentation-bunyan/src/bunyan.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { context, diag, getSpan, isSpanContextValid } from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import { BunyanInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import type * as BunyanLogger from 'bunyan';
+
+function defaultHook() {}
+
+export class BunyanInstrumentation extends InstrumentationBase<
+  typeof BunyanLogger
+> {
+  constructor(config: BunyanInstrumentationConfig = { logHook: defaultHook }) {
+    super('@opentelemetry/instrumentation-bunyan', VERSION, config);
+  }
+
+  protected init() {
+    return [
+      new InstrumentationNodeModuleDefinition<typeof BunyanLogger>(
+        'bunyan',
+        ['*'],
+        logger => {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const proto = logger.prototype as any;
```

Comment:
isn't the logger a typeof `BunyanLogger` ?

Reply:
@seemk You marked the conversion as resolved without answering nor modifying anything, is there any reason why this isn't typed ?

Reply:
> @seemk You marked the conversion as resolved without answering nor modifying anything, is there any reason why this isn't typed ?

My bad, thought I replied to this.

But the issue is that `logger` is indeed typeof `BunyanLogger`, however `prototype` does not expose `_emit` in the types thus the need for any.